### PR TITLE
Rework database and collection creation for Tigris

### LINKED ISSUE #787: Rework database and collection creation for Tigris
Currently, we try to create databases and collections in too many places. We should clean that up.

Expand `tigrisdb`. Port `CreateTableIfNotExist` and `insert` from `pg` handler. Use them where needed.

Remove our own "filter" from tigrisdb.

### PR DESCRIPTION
**Pull request by rumyantseva**
# Description

This PR closes #787.

## In this PR

- `filter` is removed
- `tigrisdb` is introduced with the `CreateCollectionIfNotExist` method, just like in `pgdb`
- Simple tests are added for `tigrisdb`
- A couple of TODOs are added because of the way how Tigris treats json schemas for objects

## Readiness checklist

* [x] I added tests for new functionality or bugfixes.
* [x] I ran `task all`, and it passed.
* [x] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handlers/tigris/tigrisdb/filter.go
Comment (rumyantseva) [inline internal/handlers/tigris/tigrisdb/filter.go:19]: @AlekSi please take a look at this file and let me know if this is how you expected me to rewrite the filter.

I kinda liked the approach Tigris offered because it was possible to have nice chains of function calls instead of lots of parentheses. But I leave the final decision to you.

File: internal/handlers/tigris/msg_delete.go
Comment (rumyantseva) [inline internal/handlers/tigris/msg_delete.go:153]: @AlekSi this is what I mean by nice chains in the comment above :)

File: internal/handlers/tigris/tigrisdb/filter.go
Comment (AlekSi) [reply internal/handlers/tigris/tigrisdb/filter.go:19]: In pgdb, we have that method:

```
func (pgPool *Pool) QueryDocuments(ctx context.Context, querier pgxtype.Querier, sp SQLParam) (<-chan FetchedDocs, error)
```

The only "filter" it accepts is SQLParam. We should have a similar API there.

Please remove all those filters.

Comment (rumyantseva) [reply internal/handlers/tigris/tigrisdb/filter.go:19]: @AlekSi hmm... This will need quite some refactoring for `MsgDelete` (as the filter is used there quite heavily), but I'm ok with it. This past of `MsgDelete` (`h.delete`) needs to be refactored anyway, as most of the `h.delete` function statements are never called.

File: internal/handlers/tigris/msg_update.go
Comment (rumyantseva) [inline internal/handlers/tigris/msg_update.go:201]: @AlekSi do you want this to be something like `fmt.Sprintf(`{"_id": {"$eq": %s}`, id)`?

Comment (AlekSi) [reply internal/handlers/tigris/msg_update.go:201]: `json.Marshal(map[string]any{...})`

Comment (rumyantseva) [reply internal/handlers/tigris/msg_update.go]: Done

**Review by AlekSi — APPROVED** [review]

**Review by w84thesun — APPROVED** [review]

**Comment by rumyantseva** [comment]
I need another review after merging the conflicts with `main` (cc @AlekSi @w84thesun @noisersup)

**Review by noisersup — APPROVED** [review]

**Review by w84thesun — APPROVED** [review]

**Review by AlekSi — APPROVED** [review]

**Pull request merged**
